### ISSUE TITLE
Reimplement the v7 policies on top of v8 APIs

### ISSUE THREAD

**Issue by martintmk** [App-vNext/Polly#1132]
Once we are satisfied with the v7 API surface we should reimplement all v7 policies on top of v8 APIs.

This will reduce the amount of code we will have to maintain for legacy v7 and it will also verify the v8 behavior.

Any regressions should be fixed and covered by tests in v8. After we are green we can also clean-up the tests in v7.

This should be done before the official v8 release to give us a great confidence in v8 quality.

**Comment by martincostello** [comment]
@martintmk Did we decide this is a _Won't do_, or should this just stay open and sit in the backlog for the future?

**Comment by martintmk** [comment]
On one hand it could simplify the v7 API, but it could also destabilize it. 

Given that v7 won't be touched very often I would say let's just do "Won't do". If we see we are making frequent changes to v7 we might re-open it and migrate the internals to use v8 API.
